src/Hyperion/Extensions/TypeEx.cs

```diff
@@ -228,9 +243,18 @@ public static string GetShortAssemblyQualifiedName(this Type self)
             return name;
         }
 
-        public static string ToQualifiedAssemblyName(string shortName)
+        public static string ToQualifiedAssemblyName(string shortName, bool ignoreAssemblyVersion)
         {
             var res = shortName.Replace(",%core%", CoreAssemblyName);
+            
+            // Strip out assembly version, if specified
+            if (ignoreAssemblyVersion)
+            {
+                var versionSubstrIndex = res.IndexOf(", Version=", StringComparison.Ordinal);
+                if (versionSubstrIndex >= 0)
+                    res = res.Remove(versionSubstrIndex);
+            }
+            
             return res;
```

Comment:
Stripping out the assembly version doesn't seem to be quite working. For a shortName: 

```System.Collections.Immutable.ImmutableDictionary`2[[System.String, mscorlib,%core%],[System.Int32, mscorlib,%core%]], System.Collections.Immutable, Version=1.2.1.0, PublicKeyToken=b03f5f7f11d50a3a```

 the output is: 

```System.Collections.Immutable.ImmutableDictionary`2[[System.String, mscorlib```

Reply:
cc @IgorFedchenko

Reply:
@ismaelhamed Huh, good catch, thanks!
Did not notice that %core% is replaced to something, that contains "Version=..." too.
Fixed, and added test case for this as well.

Reply:
Working now @IgorFedchenko , thanks!

Reply:
Actually I've encounter another case in which it fails:

```
System.Collections.Immutable.ImmutableList`1[[Foo.Bar, Foo, Version=2019.12.10.1]], System.Collections.Immutable, Version=1.2.2.0, PublicKeyToken=b03f5f7f11d50a3a
```

will also remove the `]]` along with the Version:

```
System.Collections.Immutable.ImmutableList`1[[Foo.Bar, Foo
```

Maybe too many edge cases to trying stripping the Version manually?

Reply:
@ismaelhamed Hmm, well, stripping Version as a substring was definitely bad idea.

Let's make it more careful, and strip it by regex. I will use `new Regex(@",[\s]*Version=[0-9\.]*")`, should work for any valid `Version` entry.

Except manual version stripping, the only way I know is hacking the next step - assembly loading. We could subscribe to [`AssemblyLoadContext.Resolving`](https://docs.microsoft.com/ru-ru/dotnet/api/system.runtime.loader.assemblyloadcontext.resolving?view=netcore-3.1) event and try to ignore assembly version from there. And remove subscription once type is resolved. But this is more complex approach which requires careful handling, because this event subscription is global for application and if we will miss unsubscription (i.e. due to some local handled exception), we will have all assemblies always loaded with version ignoring. Which might be not critical, but jet.

So, this is just an idea - actually cutting off `,(spaces)Version=(numbers and dots)` should cover cases I can imagine.

But @ismaelhamed thank you very much for been so attentive!

Reply:
@IgorFedchenko I was thinking it should probably be a good idea to make that Regex static, and I just realized there's already a `cleanAssemblyVersionRegex` in this file.

Reply:
@ismaelhamed Indeed, there is an existing regex that is used to strip assembly version before serializing types. Thanks again, this looks even better now :)